app/lib/link_details_extractor.rb

```diff
@@ -294,7 +295,7 @@ def detector
   def html_entities_decode(string)
     return if string.nil?
 
-    unicode_string = string.encode('UTF-8')
+    unicode_string = string.to_s.encode('UTF-8')
```

Comment:
I assume this is here as a fallback, so it is processed like it was before, but not used for the array-of-authors case?

Reply:
Exactly! I started out with this as a "fix" and then noticed the unfortunate display of "arrayed" author names. I thought it would make no sense to remove this afterwards, as this basically brings back the old behavior before #30929 for any other edge cases that might still be out there.